tests/conftest.py

```
@@ -74,6 +78,22 @@ def wait_for_healthy(self, timeout=30):
 
+    def wait_for_legacy(self, timeout=30):
+        """/_health only answers for the HTTP listener. The legacy port is a
+        separately spawned task, so a test that connects to it straight after
+        start() races the bind -- which is why the first legacy test failed on
+        every run with ConnectionRefusedError."""
+        deadline = time.time() + timeout
+        last_err = None
+        while time.time() < deadline:
+            try:
+                socket.create_connection(("127.0.0.1", self.legacy_port), timeout=1).close()
+                return
```

Ruff (0.16.1)

[warning] Consider moving this statement to an `else` block

(TRY300)